src/GraphQLParser/Visitors/IWriteContext.cs

```diff
@@ -0,0 +1,12 @@
+using System.Collections.Generic;
+using System.IO;
+
+namespace GraphQLParser.Visitors
+{
+    public interface IWriteContext : IVisitorContext
+    {
+        TextWriter Writer { get; }
+
+        Stack<AST.ASTNode> Parent { get; }
+    }
```

Comment:
```suggestion

        CancellationToken CancellationToken { get; }
    }
```
I suggest adding cancellation token support

Reply:
done